test/e2e/autoscale_test.go

```diff
@@ -119,9 +120,10 @@ func generateTraffic(
 		select {
 		case <-stopChan:
 			ctx.t.Log("Stopping generateTraffic")
-			if successfulRequests != totalRequests {
-				return fmt.Errorf("error making requests for scale up: total = %d, errors = %d, expected 0",
-					totalRequests, totalRequests-successfulRequests)
+			successRate := float32(successfulRequests)/float32(totalRequests)
```

Comment:
are we sure there is never a test with 0 totalRequests?

Reply:
+1.
Also why float32? Why do you hate precision ;-)

Reply:
This test runs at around 600 QPS, in no-mesh we don't see errors, in mesh I think we would need to see what's the bottleneck but my guess is some Envoy are saturated.
But the more important point here is that this test is designed to test the scaling behavior of the autoscaler, not to be a performance test, as such I believe seeing 0 error is not key here.

Reply:
No, the question was that you might get a division by zero.

Reply:
Ah, totally misunderstood the comment.